Mod poly and mpn_mod random generation (#2448)
flintlib/flint: doc/source/fmpz_mod_poly.rst

    set to a random monic irreducible polynomial.

.. function:: void fmpz_mod_poly_rand(fmpz_mod_poly_t f, flint_rand_t state, slong len, const fmpz_mod_ctx_t ctx)

    Sets `f` to a random polynomial with up to the given length and where
    each coefficient has up to the given number of bits. The coefficients

>     are uniformly generated random numbers in `[0, n)`, where `n` is the modulus given by the context `ctx`.

Reviewer: I would suggest to copy-paste the doc of e.g. `nmod_poly`:

```suggestion
    Generates a random polynomial with length up to `len`, with uniformly chosen coefficients.
```

.. function:: void fmpz_mod_poly_rand_monic(fmpz_mod_poly_t f, flint_rand_t state, slong len, const fmpz_mod_ctx_t ctx)

    Sets `f` to a random monic polynomial with up to the given length and where
    each coefficient has up to the given number of bits. The coefficients

>     are uniformly generated random numbers in `[0, n)`, where `n` is the modulus given by the context `ctx`.

Reviewer: I would suggest to copy-paste the doc of e.g. `nmod_poly`:

```suggestion
    Generates a random monic polynomial with length `len`, with uniformly chosen coefficients.
```

.. function:: void fmpz_mod_poly_rand_irreducible(fmpz_mod_poly_t f, flint_rand_t state, slong len, const fmpz_mod_ctx_t ctx)

    Sets `f` to a random irreducible polynomial with up to the given length and where each coefficient has up to the given number of bits. The coefficients

>     are uniformly generated random numbers in `[0, n)`, where `n` is the modulus given by the context `ctx`.

Reviewer: As above, I would suggest something more concise (and there is nothing about number of bits here):

```suggestion
    Generates a random irreducible polynomial with up to the given length and where the coefficients
    are generated uniformly.
```
